test/framework/toolchain.py

```diff
@@ -381,6 +381,30 @@ def test_comp_family(self):
         tc.prepare()
         self.assertEqual(tc.comp_family(), "GCC")
 
+    def test_mpi_family(self):
+        """Test determining MPI family."""
+        # check subtoolchain w/o MPI
+        tc = self.get_toolchain("GCC", version="4.7.2")
+        tc.prepare()
+        self.assertEqual(tc.mpi_family(), None)
+        modules.modules_tool().purge()
+
+        # check full toolchain including MPI
+        tc = self.get_toolchain("goalf", version="1.1.0-no-OFED")
+        tc.prepare()
+        self.assertEqual(tc.mpi_family(), "OpenMPI")
+        modules.modules_tool().purge()
+
+        # check another one
+        tmpdir, imkl_module_path, imkl_module_txt = self.setup_sandbox_for_intel_fftw()
+        tc = self.get_toolchain("ictce", version="4.1.13")
+        tc.prepare()
+        self.assertEqual(tc.mpi_family(), "IntelMPI")
+
+        # cleanup
+        shutil.rmtree(tmpdir)
+        open(imkl_module_path, 'w').write(imkl_module_txt)
```

Comment:
use `write_file` here, and please also include this in the other places where the `imkl` module is being fiddled with...

the problem is: if the test fails, the `imkl` module will not be restored (this issue was already there, it's not something you're causing)